src/websocket/server.cc

```diff
@@ -126,6 +130,30 @@ future<> connection::process() {
 
 static std::string sha1_base64(std::string_view source) {
     unsigned char hash[20];
+
+#ifdef SEASTAR_USE_OPENSSL
```

Comment:
Could we instead make the function an external/internal-header-decl, and move the full impl into gnutls/openssl compilation units? This just adds to an already bad separation of concern/api exposure.